{"parents": [{"link": "../", "title": "Release notes"}], "prev": {"link": "../2.2.4/", "title": "Django 2.2.4 release notes"}, "next": {"link": "../2.2.2/", "title": "Django 2.2.2 release notes"}, "title": "Django 2.2.3 release notes", "meta": {}, "body": "<div class=\"section\" id=\"s-django-2-2-3-release-notes\">\n<span id=\"django-2-2-3-release-notes\"></span><h1>Django 2.2.3 release notes<a class=\"headerlink\" href=\"#django-2-2-3-release-notes\" title=\"Permalink to this headline\">\u00b6</a></h1>\n<p><em>July 1, 2019</em></p>\n<p>Django 2.2.3 fixes a security issue and several bugs in 2.2.2. Also, the latest\nstring translations from Transifex are incorporated.</p>\n<div class=\"section\" id=\"s-cve-2019-12781-incorrect-http-detection-with-reverse-proxy-connecting-via-https\">\n<span id=\"cve-2019-12781-incorrect-http-detection-with-reverse-proxy-connecting-via-https\"></span><h2>CVE-2019-12781: Incorrect HTTP detection with reverse-proxy connecting via HTTPS<a class=\"headerlink\" href=\"#cve-2019-12781-incorrect-http-detection-with-reverse-proxy-connecting-via-https\" title=\"Permalink to this headline\">\u00b6</a></h2>\n<p>When deployed behind a reverse-proxy connecting to Django via HTTPS,\n<a class=\"reference internal\" href=\"../../ref/request-response/#django.http.HttpRequest.scheme\" title=\"django.http.HttpRequest.scheme\"><code class=\"xref py py-attr docutils literal notranslate\"><span class=\"pre\">django.http.HttpRequest.scheme</span></code></a> would incorrectly detect client\nrequests made via HTTP as using HTTPS. This entails incorrect results for\n<a class=\"reference internal\" href=\"../../ref/request-response/#django.http.HttpRequest.is_secure\" title=\"django.http.HttpRequest.is_secure\"><code class=\"xref py py-meth docutils literal notranslate\"><span class=\"pre\">is_secure()</span></code></a>, and\n<a class=\"reference internal\" href=\"../../ref/request-response/#django.http.HttpRequest.build_absolute_uri\" title=\"django.http.HttpRequest.build_absolute_uri\"><code class=\"xref py py-meth docutils literal notranslate\"><span class=\"pre\">build_absolute_uri()</span></code></a>, and that HTTP\nrequests would not be redirected to HTTPS in accordance with\n<a class=\"reference internal\" href=\"../../ref/settings/#std-setting-SECURE_SSL_REDIRECT\"><code class=\"xref std std-setting docutils literal notranslate\"><span class=\"pre\">SECURE_SSL_REDIRECT</span></code></a>.</p>\n<p><code class=\"docutils literal notranslate\"><span class=\"pre\">HttpRequest.scheme</span></code> now respects <a class=\"reference internal\" href=\"../../ref/settings/#std-setting-SECURE_PROXY_SSL_HEADER\"><code class=\"xref std std-setting docutils literal notranslate\"><span class=\"pre\">SECURE_PROXY_SSL_HEADER</span></code></a>, if it is\nconfigured, and the appropriate header is set on the request, for both HTTP and\nHTTPS requests.</p>\n<p>If you deploy Django behind a reverse-proxy that forwards HTTP requests, and\nthat connects to Django via HTTPS, be sure to verify that your application\ncorrectly handles code paths relying on <code class=\"docutils literal notranslate\"><span class=\"pre\">scheme</span></code>, <code class=\"docutils literal notranslate\"><span class=\"pre\">is_secure()</span></code>,\n<code class=\"docutils literal notranslate\"><span class=\"pre\">build_absolute_uri()</span></code>, and <code class=\"docutils literal notranslate\"><span class=\"pre\">SECURE_SSL_REDIRECT</span></code>.</p>\n<div class=\"section\" id=\"s-bugfixes\">\n<span id=\"bugfixes\"></span><h3>Bugfixes<a class=\"headerlink\" href=\"#bugfixes\" title=\"Permalink to this headline\">\u00b6</a></h3>\n<ul class=\"simple\">\n<li>Fixed a regression in Django 2.2 where <a class=\"reference internal\" href=\"../../ref/models/querysets/#django.db.models.Avg\" title=\"django.db.models.Avg\"><code class=\"xref py py-class docutils literal notranslate\"><span class=\"pre\">Avg</span></code></a>,\n<a class=\"reference internal\" href=\"../../ref/models/querysets/#django.db.models.StdDev\" title=\"django.db.models.StdDev\"><code class=\"xref py py-class docutils literal notranslate\"><span class=\"pre\">StdDev</span></code></a>, and <a class=\"reference internal\" href=\"../../ref/models/querysets/#django.db.models.Variance\" title=\"django.db.models.Variance\"><code class=\"xref py py-class docutils literal notranslate\"><span class=\"pre\">Variance</span></code></a>\ncrash with <code class=\"docutils literal notranslate\"><span class=\"pre\">filter</span></code> argument (<a class=\"reference external\" href=\"https://code.djangoproject.com/ticket/30542\">#30542</a>).</li>\n<li>Fixed a regression in Django 2.2.2 where auto-reloader crashes with\n<code class=\"docutils literal notranslate\"><span class=\"pre\">AttributeError</span></code>, e.g. when using <code class=\"docutils literal notranslate\"><span class=\"pre\">ipdb</span></code> (<a class=\"reference external\" href=\"https://code.djangoproject.com/ticket/30588\">#30588</a>).</li>\n</ul>\n</div>\n</div>\n</div>\n", "metatags": "", "rellinks": [["genindex", "General Index", "I", "index"], ["py-modindex", "Python Module Index", "", "modules"], ["releases/2.2.2", "Django 2.2.2 release notes", "N", "next"], ["releases/2.2.4", "Django 2.2.4 release notes", "P", "previous"]], "sourcename": "releases/2.2.3.txt", "toc": "<ul>\n<li><a class=\"reference internal\" href=\"#\">Django 2.2.3 release notes</a><ul>\n<li><a class=\"reference internal\" href=\"#cve-2019-12781-incorrect-http-detection-with-reverse-proxy-connecting-via-https\">CVE-2019-12781: Incorrect HTTP detection with reverse-proxy connecting via HTTPS</a><ul>\n<li><a class=\"reference internal\" href=\"#bugfixes\">Bugfixes</a></li>\n</ul>\n</li>\n</ul>\n</li>\n</ul>\n", "display_toc": true, "page_source_suffix": ".txt", "current_page_name": "releases/2.2.3", "sidebars": ["localtoc.html", "relations.html", "sourcelink.html", "searchbox.html"], "customsidebar": null, "favicon_url": null, "logo_url": null, "alabaster_version": "0.7.13", "include_console_assets": false}
